tests/unit/testsuite/Bolt/Boltpay/Block/Checkout/BoltpayTest.php

```diff
@@ -0,0 +1,437 @@
+                input.setAttribute('type', 'hidden');
+                input.setAttribute('name', 'bolt_reference');
+                input.setAttribute('value', transaction.reference);
+                document.getElementById('edit_form').appendChild(input);
+
+                // order and order.submit should exist for admin
+                if ((typeof order !== 'undefined' ) && (typeof order.submit === 'function')) {
+                    order_completed = true;
+                    callback();
+                }
+            }";
+
+        $result = $this->currentMock->buildOnSuccessCallback($successCustom, $checkoutType);
+
+        $this->assertEquals($onSuccessCallback, $result);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function testBuildOnCloseCallback()
+    {
+        $successUrl = Mage::getUrl('checkout/onepage/success');
+        $checkoutType = Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_ONE_PAGE;
+        $closeCustom = '';
+
+        $expect = "if (typeof bolt_checkout_close === 'function') {
+                   // used internally to set overlay in firecheckout
+                   bolt_checkout_close();
+                }
+                if (order_completed) {
+                   location.href = '$successUrl';
+            }";
+
+        $result = $this->currentMock->buildOnCloseCallback($closeCustom, $checkoutType);
+
+        $this->assertEquals($expect, $result);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function testBuildOnCloseCallbackIfAdminArea()
+    {
+        $checkoutType = Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_ADMIN;
+        $closeCustom = '';
+
+        $expect = "if (order_completed && (typeof order !== 'undefined' ) && (typeof order.submit === 'function')) {
+                $closeCustom
+                var bolt_hidden = document.getElementById('boltpay_payment_button');
+                bolt_hidden.classList.remove('required-entry');
+                order.submit();
+             }";
+
+        $result = $this->currentMock->buildOnCloseCallback($closeCustom, $checkoutType);
+
+        $this->assertEquals($expect, $result);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function testGetTrackJsUrl()
+    {
+        $this->app->getStore()->setConfig('payment/boltpay/test', false);
+
+        $expect = Bolt_Boltpay_Block_Checkout_Boltpay::JS_URL_PROD . "/track.js";
+
+        $result = $this->currentMock->getTrackJsUrl();
+
+        $this->assertEquals($expect, $result);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function testGetTrackJsUrlIfTestMode()
+    {
+        $this->app->getStore()->setConfig('payment/boltpay/test', true);
+
+        $expect = Bolt_Boltpay_Block_Checkout_Boltpay::JS_URL_TEST . "/track.js";
+
+        $result = $this->currentMock->getTrackJsUrl();
+
+        $this->assertEquals($expect, $result);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function testGetCssSuffix()
```

Comment:
IMO we don't need unit test for oneline getter just returns constant. 
Maintenance burden would be more than benefit.